From 952f75a370060087373ff0cf0c10d66334a9c47f Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 24 Oct 2018 17:11:09 +0000
Subject: [PATCH 0850/1000] MANTIS 0045259 extension: MAT_CLA: HTML5: replace
 CORBA by WS

mstarter.exe
V1.1.1.13
- use mText::getConfigValues for variable substitution (TRACES.TrcServ.TraceDirectory)


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1386 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/Controller/inc/mstarter_def.h          | 11 ++++++++---
 .../Massai/cpp/Controller/src/mstarter_config.cpp     | 11 ++++++-----
 2 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Controller/inc/mstarter_def.h b/SelfServiceCommon/Massai/cpp/Controller/inc/mstarter_def.h
index bf9f0537..a1f43110 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/inc/mstarter_def.h
+++ b/SelfServiceCommon/Massai/cpp/Controller/inc/mstarter_def.h
@@ -1,5 +1,5 @@
-#define FILE_VER_BIN     1,1,1,12
-#define FILE_VER_STR     "FileVersion", "1.1.1.12"
+#define FILE_VER_BIN     1,1,1,13
+#define FILE_VER_STR     "FileVersion", "1.1.1.13"
 
 /*!
 \file
@@ -7,12 +7,17 @@
 
 \page mstarter.exe Release Notes - mstarter.exe - process manager
 
-@version 1.1.1.12
+@version 1.1.1.13
 
 \section change1 Change history
 
 mstarter.exe
 
+\version 1.1.1.13
+\li 24.10.2018
+\li extension - SteT Mantis 0045259 MAT_CLA: HTML5: replace CORBA by WS
+- use mText::getConfigValues for variable substitution (TRACES.TrcServ.TraceDirectory)
+
 \version 1.1.1.12
 \li 26.07.2018
 \li change - Mantis 0042165: mstarter creates configuration entries PLATFORM.EXEC.*!java with wrong description
diff --git a/SelfServiceCommon/Massai/cpp/Controller/src/mstarter_config.cpp b/SelfServiceCommon/Massai/cpp/Controller/src/mstarter_config.cpp
index 607cc742..df961a1e 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/src/mstarter_config.cpp
+++ b/SelfServiceCommon/Massai/cpp/Controller/src/mstarter_config.cpp
@@ -62,7 +62,7 @@ namespace
       "  close_wnd = \"window name\" - specifies the window name used for process shutdown.\n"
       "  close_wndclass = \"window class\" - specifies the window class used for process shutdown.\n"
       "The following substitution environment variables are supported: "
-      "$PROGRAMFILES, $SYSTEMROOT, $SYSTEMDRIVE";  
+      "$PROGRAMFILES, $SYSTEMROOT, $SYSTEMDRIVE";
 
   //  base class for config items
   class ConfigItemBase
@@ -91,9 +91,9 @@ namespace
     public:
       ConfigExec(const std::string& item);
       TextList getValue() const;
-      virtual void valueChanged();      
+      virtual void valueChanged();
   };
-  
+
   typedef boost::shared_ptr<ConfigExec> ConfigExecPtr;
   typedef std::map<std::string, ConfigExecPtr> ExecMap;
 
@@ -162,12 +162,12 @@ namespace
     {
       if(m_item_.find(configJavaPostfix) == m_item_.length()-configJavaPostfix.length()-1)
       {
-        description = javaExecItemDesc;      
+        description = javaExecItemDesc;
       }
     }
     catch(...)
     {
-      std::cout << "mstarter.exe: exception in ConfigExec::specific_description" << std::endl;      
+      std::cout << "mstarter.exe: exception in ConfigExec::specific_description" << std::endl;
     }
     return description;
   }
@@ -474,6 +474,7 @@ namespace mStarter
     m_substituteEnv.SetBaseEnvVars();
     m_substituteEnv.SetEnvVars(mText::getSystemEnvironment(m_trc));
     m_substituteEnv.SetEnvVars(mText::getConfigVariables(configUser, m_trc));
+    m_substituteEnv.SetEnvVars(mText::getConfigValues(configUser, m_trc));
   }
 
   void Config::setListener(mSysProcess::ProcessList* pListener)
-- 
2.41.0.windows.1

